### PR TITLE
8319993: Update Unicode Data Files to 16.0.0

### PR DESCRIPTION
**Pull request by naotoj**
Updates the JDK to use the latest Unicode 16.0.0, which also updates the ICU4J along with it (8319992: Update ICU4J to Version 76.1). The corresponding CSR has already been approved.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8339568](https://bugs.openjdk.org/browse/JDK-8339568) to be approved

### Warnings
&nbsp;⚠️ Patch contains a binary file (src/java.base/share/classes/jdk/internal/icu/impl/data/icudt76b/nfc.nrm)
&nbsp;⚠️ Patch contains a binary file (src/java.base/share/classes/jdk/internal/icu/impl/data/icudt76b/nfkc.nrm)
&nbsp;⚠️ Patch contains a binary file (src/java.base/share/classes/jdk/internal/icu/impl/data/icudt76b/ubidi.icu)
&nbsp;⚠️ Patch contains a binary file (src/java.base/share/classes/jdk/internal/icu/impl/data/icudt76b/uprops.icu)

### Issues
 * [JDK-8319993](https://bugs.openjdk.org/browse/JDK-8319993): Update Unicode Data Files to 16.0.0 (**Enhancement** - P3)
 * [JDK-8319992](https://bugs.openjdk.org/browse/JDK-8319992): Update ICU4J to Version 76.1 (**Enhancement** - P4)
 * [JDK-8339568](https://bugs.openjdk.org/browse/JDK-8339568): Update Unicode Data Files to 16.0.0 (**CSR**)

### Reviewers
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21771/head:pull/21771` \
`$ git checkout pull/21771`

Update a local copy of the PR: \
`$ git checkout pull/21771` \
`$ git pull https://git.openjdk.org/jdk.git pull/21771/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21771`

View PR using the GUI difftool: \
`$ git pr show -t 21771`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21771.diff">https://git.openjdk.org/jdk/pull/21771.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21771#issuecomment-2489340980)
</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/util/regex/Grapheme.java [hunk 1]
```diff
@@ -267,7 +267,10 @@ static int getType(int cp) {
             if (cp >= 0xA960 && cp <= 0xA97C)
                 return L;
             //  hangul jamo_extended B
-            if (cp >= 0xD7B0 && cp <= 0xD7C6)
+            //  Kirat Rai vowel sign
+            if (cp >= 0xD7B0 && cp <= 0xD7C6 ||
+                cp == 0x16D63 ||
```
Comment (justin-curtis-lu) [inline src/java.base/share/classes/jdk/internal/util/regex/Grapheme.java:272]: LGTM as this matches up with https://www.unicode.org/reports/tr29/tr29-45.html#V and the `case 0x113D1:` lines up with _TULU-TIGALARI REPHA_.

File: src/java.base/share/classes/jdk/internal/util/regex/Grapheme.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public final class Grapheme {
      * for the extended grapheme cluster boundary rules. The following implementation
      * is based on the annex for Unicode version 15.1.
```
[the remark this answers is not in the document]
Comment (naotoj) [reply src/java.base/share/classes/jdk/internal/util/regex/Grapheme.java]: Good catch! Corrected.

**Review by justin-curtis-lu — APPROVED** [review]

**Review by JoeWang-Java — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Comment by naotoj** [comment]
Thanks for the reviews!
/integrate

**Pull request closed without merging**